[Ang Bin En] iP

Chinorea opened the pull request.

Chinorea committed:
Level-0

Chinorea committed:
Level-1

Added BotText Class for any text related message for the Bot.
Added getInput() function to Echo Messages

Chinorea committed:
Level-2

Added add task functionality
Added List task functionality

Chinorea committed:
Level-3

Add Task.java class file for task object representation [A-classes]
Add mark as done feat with markList and unmarkList functions

Chinorea committed:
Level-4 + A-Inheritance

Add ToDo feat with Todo class
Add Deadline feat with Deadline class
Add Event feat with Event class
Add Inheritance for multiple task type support

Chinorea committed:
A-TextUiTesting

Add Automated Text Ui Testing to Level-4 feat of iP

Chinorea committed:
A-CodeQuality

Improve code quality
Peform regression testing with code changes

Alfred-Goh02 commented on src/main/java/Ellio.java:
```diff
@@ -0,0 +1,137 @@
+        if(input.equals("list")){
+            printList();
+        }
+        else if(input.startsWith("mark")){
```
Consider using the K&R style while typing if else statements

Alfred-Goh02 commented on src/main/java/Ellio.java:
```diff
@@ -0,0 +1,137 @@
+    public static void main(String[] args) {
+        startProgram();
+        getInput();
+        endProgram();
```
Remember to close your scanner

Alfred-Goh02 commented on src/main/java/Ellio.java:
```diff
@@ -0,0 +1,137 @@
+        Todo newTodo = new Todo(description);
+        listTasks[numberTask] = newTodo;
+        numberTask++;
+        System.out.println(BotText.lineBorder + "Got it. I've added this task:\n  " + newTodo.getTask());
+        System.out.println("Now you have " + numberTask + " tasks in the list.\n" + BotText.lineBorder);
```
Can consider creating a printing class and use a toString method for your individual class, and print using your printing class methods

Alfred-Goh02 commented on src/main/java/Ellio.java:
```diff
@@ -0,0 +1,137 @@
+
+public class Ellio {
+
+    public static final int DEADLINE_INPUT_SPACING = 9;
+    public static final int EVENT_INPUT_SPACING = 6;
+    public static final int MAX_TASK = 100;
```
Good example of naming constants properly in java

yakultbottle replied:
I like that your constants follow the coding standards

yakultbottle commented on src/main/java/BotText.java:
```diff
@@ -0,0 +1,8 @@
+public class BotText {
+    public static String lineBorder = "______________________________________\n";
+    public static String messageWelcome = "Hello! I'm Ellio\nWhat can I do for you?\n";
+    public static String messageGoodbye = "Bye. Hope to see you again soon!\n";
+    public static String messageList = "Here are the tasks in your list:\n";
+    public static String messageMarked = "Nice! I've marked this task as done:\n";
+    public static String messageUnmark = "OK, I've marked this task as not done yet:\n";
```
Should this have empty lines between methods for readability?

yakultbottle commented on src/main/java/Ellio.java:
```diff
@@ -0,0 +1,137 @@
+        return new Event(description, eventStart, eventEnd);
+    }
+
+    public static void markList(int index){
+        listTasks[index-1].markTaskAsDone();
+        System.out.println(BotText.lineBorder + BotText.messageMarked + "  " + listTasks[index-1].getTask() + "\n" + BotText.lineBorder);
+    }
+
+    public static void unmarkList(int index){
+        listTasks[index-1].unmarkTaskAsDone();
+        System.out.println(BotText.lineBorder + BotText.messageUnmark + "  " + listTasks[index-1].getTask() + "\n" + BotText.lineBorder);
+    }
```
Should List be abstracted into its own class?

C2linaung commented on src/main/java/Ellio.java:
```diff
@@ -0,0 +1,137 @@
+    }
+
+    public static void printInput(String input){
+        if(input.equals("list")){
```
Would it be better to use switch cases here?

yakultbottle commented on src/main/java/Ellio.java:
```diff
@@ -0,0 +1,137 @@
+        }
+    }
+
+    public static void printInput(String input){
+        if(input.equals("list")){
+            printList();
+        }
+        else if(input.startsWith("mark")){
+            String indexNum = getTaskIndex(input);
+            markList(Integer.parseInt(indexNum));
+        }
+        else if(input.startsWith("unmark")){
+            String indexNum = getTaskIndex(input);
+            unmarkList(Integer.parseInt(indexNum));
+        }
+        else if(input.startsWith("todo")){
+            addToDo(input);
+        }
+        else if(input.startsWith("deadline")){
+            addDeadline(input);
+        }
+        else if(input.startsWith("event")){
+            addEvent(input);
+        }
+    }
```
Should this be a switch statement instead of if-else?

yakultbottle commented on src/main/java/Ellio.java:
```diff
@@ -0,0 +1,137 @@
+        return new Deadline(description, deadline);
+    }
+
+    public static void addEvent(String line){
+        Event newEvent = formatEvent(line);
+        listTasks[numberTask] = newEvent;
+        numberTask++;
+        System.out.println(BotText.lineBorder + "Got it. I've added this task:\n  " + newEvent.getTask());
+        System.out.println("Now you have " + numberTask + " tasks in the list.\n" + BotText.lineBorder);
+    }
```
Should this be abstracted into another class?

yakultbottle commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+public class Event extends Task{
+
+    protected String eventStartTime;
+    protected String eventEndTime;
```
Should these variables be declared with "final"?

C2linaung commented on src/main/java/Ellio.java:
```diff
@@ -0,0 +1,137 @@
+import java.util.Scanner;
+
+public class Ellio {
```
I think your spacing and indents are pretty good. really like it there

C2linaung commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,31 @@
+    }
+
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
```
is the comment here necessary?

C2linaung commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,15 @@
+    }
+
+    public String getTask(){
+        return "[E]" + super.getTask() + " (" + eventStartTime + " " +eventEndTime + ")";
```
can take note of the spacing here

C2linaung commented on src/main/java/Ellio.java:
```diff
@@ -0,0 +1,137 @@
+
+public class Ellio {
+
+    public static final int DEADLINE_INPUT_SPACING = 9;
```
constants are in caps so thats good.

C2linaung reviewed (commented):
good job on the code generally. mostly followed the convention

Chinorea committed:
Changed UI-Testing for tutorial

Chinorea committed:
Add Level-5 Increment and A-Exception Extension

Added Error Handling for the following cases:
Unknown Error input
Empty List
Invalid Index (for Mark & Unmark commands)
Invalid Deadline format (Missing fields)
Invalid Event format (Missing fields

Chinorea committed:
Merge branch 'branch-Level-5'

Chinorea committed:
A-Packages

Refactored exisiting files into seperate packages.
Consolidated all expception class file into one common file

Chinorea committed:
Merge branch 'branch-A-Packages'

Chinorea committed:
A-Collection

Fix Tasklist to an Arraylist.

Chinorea committed:
Level-6

Added delete functions with error detection

Chinorea committed:
Level-7

Add Save function

Chinorea committed:
Merge branch 'branch-Level-6'

Chinorea committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/ellio/Ellio.java

Chinorea committed:
Add delete func integration with Save

Add delete function compatability with Save function

Chinorea committed:
A-Jar

Create Jar file manifesto

okkhoy commented on src/main/java/ellio/BotText.java:
```diff
@@ -0,0 +1,18 @@
+    public static String messageInvalidEventEndFormat = "The format used for your end time is incorrect. Initialise end event time with a /to\n";
+    public static String messageInvalidEventTimeFormat = "Missing Start or End timing detected. Remember to add /from and /to to your event\n";
+    public static String messageInvalidDeadlineDateFormat = "The format used for your date is incorrect. Initialise date with a /by\n";
+    public static String messageMissingDeadlineDate = "Missing deadline date detected. Remember to add /by to your deadline\n";
```
Potentially make all these constants?

okkhoy commented on src/main/java/ellio/Ellio.java:
```diff
@@ -0,0 +1,312 @@
+        if(!saveFile.exists()){
+            createFile();
+        }
+        else{
```
coding standard violation

okkhoy commented on src/main/java/ellio/Ellio.java:
```diff
@@ -0,0 +1,312 @@
+        for(int i = 1; i < listTasks.size(); i++){
+            updatedTasksForSaveFile += System.lineSeparator() + listTasks.get(i).getSaveFileTask();
+        }
+        try{
```
spacing problem

okkhoy commented on src/main/java/ellio/Ellio.java:
```diff
@@ -0,0 +1,312 @@
+        saveTask(newTodo.getSaveFileTask());
+    }
+
+    public static void addDeadline(String line){
```
all non-trivial methods need header comments

okkhoy commented on src/main/java/ellio/Ellio.java:
```diff
@@ -0,0 +1,312 @@
+            System.out.println("Now you have " + numberTask + " tasks in the list.\n" + BotText.lineBorder);
+            saveTask(newEvent.getSaveFileTask());
+        }
+        catch (EllioExceptions e){
```
spacing issue. this seems like a common problem across the code. please fix it

okkhoy commented on src/main/java/ellio/Ellio.java:
```diff
@@ -0,0 +1,312 @@
+        catch (EllioExceptions e){
+            System.out.println(e.getMessage());
+        }
+        catch (StringIndexOutOfBoundsException e){
```
coding standard violation

okkhoy commented on src/main/java/ellio/Ellio.java:
```diff
@@ -0,0 +1,312 @@
+            throw new IndexOutOfBoundsException();
+        }
+        listTasks.get(index-1).unmarkTaskAsDone();
+        System.out.println(BotText.lineBorder + BotText.messageUnmark + "  " + listTasks.get(index-1).getTask() + "\n" + BotText.lineBorder);
```
particularly long line

okkhoy commented on src/main/java/ellio/EllioExceptions.java:
```diff
@@ -0,0 +1,25 @@
+            super(BotText.lineBorder + BotText.messageInvalidDeadlineDateFormat + BotText.lineBorder);
+        }
+    }
+    public static class WrongEventEndFormatException extends EllioExceptions{
```
vertical spacing for clarity

okkhoy commented on src/main/java/ellio/task/Task.java:
```diff
@@ -0,0 +1,48 @@
+        }
+        return "| " + markState + " | " + description;
+    }
+    //...
```
redundant/dead code

okkhoy reviewed (commented):
There are quite a few coding standard violations in the code. Please fix them before the final submission

Chinorea committed:
A-MoreOOP

Add the following classes through Refactorisation
Ui: Handling of UI interfaces
Command (and the list of valid commands)
Storage: Handling of Storage Related Functions
Parser: Intepretting User Command
TaskList: ListArray Functions & Object

Chinorea committed:
Level-8

Add Dates & time with LocalDate java library
Minimal Stretch Goal Achieved

Chinorea committed:
Level-9

Add find Task function

Chinorea committed:
Merge pull request #1 from Chinorea/branch-Level-9

Level-9

Chinorea committed:
Merge pull request #2 from Chinorea/branch-Level-8

Level-8

Chinorea committed:
Merge branch 'master' of https://github.com/Chinorea/ip

* 'master' of https://github.com/Chinorea/ip:
  Level-9
  Level-8

Chinorea committed:
A-JavaDoc

Add JavaDoc Header Comments to majority of non-private classes/methods

Chinorea committed:
Update A-JavaDoc

Added new JavaDoc Comments

Chinorea committed:
Merge pull request #3 from Chinorea/branch-A-JavaDoc

Branch A-java doc

Chinorea committed:
A-UserGuide

Update UserGuide

Chinorea committed:
Update ReadMe Contents

Chinorea committed:
Add Help Message & Intro

Chinorea committed:
Fix Empty List Storage issue.

Chinorea committed:
Add DateTime feature to event class

Chinorea committed:
Update Readme Doc

Chinorea committed:
Merge pull request #4 from Chinorea/branch-Level-8-Stretch

Branch level 8 stretch

Chinorea committed:
Fix Text-Ui-Test

Chinorea committed:
Merge branch 'branch-Level-8-Stretch'

Chinorea committed:
Merge branch 'master' of https://github.com/Chinorea/ip

Chinorea committed:
Fix negative index bugs

Chinorea committed:
Fix invalid index commands (string user input)
